main.go

```diff
@@ -22,51 +22,73 @@ func main() {
 
 	// Iterate over each pull request
 	for _, pr := range prs {
-		fmt.Printf("PR #%d: %s\n", pr.GetNumber(), pr.GetTitle())
-
-		fmt.Printf("Getting diff for PR #%d\n", pr.GetNumber())
-		diff, err := client.GetPrDiff(pr)
-		if err != nil {
-			log.Fatalf("Failed to get raw diff: %v", err)
+		comments := reviewPR(pr, client)
+		commitId := pr.GetHead().GetSHA()
+		for _, comment := range comments {
+			// post to github
+			fmt.Printf("Posting comment to PR #%d\n", pr.GetNumber())
+			ghComment := client.ParsedCommentToGithubComment(comment, commitId)
+			err := client.PostComment(pr, ghComment)
+			if err != nil {
+				log.Fatalf("Failed to post comment: %v", err)
+			}
 		}
+	}
+}
+
+func reviewPR(pr *github.PullRequest, client github_client.GithubRepoClient) []*comments.Comment {
+	fmt.Printf("PR #%d: %s\n", pr.GetNumber(), pr.GetTitle())
+
+	fmt.Printf("Getting diff for PR #%d\n", pr.GetNumber())
+	diff, err := client.GetPrDiff(pr)
+	if err != nil {
+		log.Fatalf("Failed to get raw diff: %v", err)
+	}
+
+	diffTransformer := transformer.DiffTransformer{}
+	diffTransformer.Transform(diff)
 
-		diffTransformer := transformer.DiffTransformer{}
-		diffTransformer.Transform(diff)
+	fileSegments := diffTransformer.GetFileSegments()
+	allComments := []*comments.Comment{}
+	// failedComments := []string{}
 
-		segments := diffTransformer.GetSegments()
-		allComments := []string{}
-		fmt.Printf("Getting commments for %d segments\n", len(segments))
+	fmt.Printf("Getting comments for %d segments\n", len(fileSegments))
+	for filename, segments := range fileSegments {
 		for _, segment := range segments {
 			comment := retrieveComments(segment)
-			allComments = append(allComments, comment)
+			if comment == nil {
+				// failedComments = append(failedComments, segment)
+				continue
+			}
+			zippedComments, err := comments.ZipComment(segment, *comment, filename)
+			if err != nil {
+				log.Fatalf("Failed to zip comment: %v", err)
+			}
+			allComments = append(allComments, zippedComments...)
+			if len(allComments) >= 1 {
+				break
+			}
+		}
+		if len(allComments) >= 1 {
+			break
 		}
-
-		writeResults(allComments)
 	}
+
+	return allComments
 }
 
-func retrieveComments(segment string) string {
+func retrieveComments(segment string) *string {
 	openAiClient := openai.OpenAiClient{}
 	openAiClient.Init()
 	completion, err := openAiClient.GetCompletion(segment)
 	if err != nil {
 		fmt.Printf("Failed to get completion: %v\n", err)
 	}
 
-	fmt.Println("********************")
```

Comment:
Bug: the loop breaking condition is incorrect. It will only break out of the inner loop after the comments slice has been appended to at least once, but will only break out of the outer loop if the slice has at least one element. This can lead to infinite loops.